### ISSUE TITLE
haxe 3.2 don't compile

### ISSUE THREAD

**Issue by francescoagati** [jasononeil/dodrugs#7]
with haxe 3.2 i get this error

```
test/Example.hx:13: characters 9-24 : Class<dodrugs.Injector> has no field create
test/Example.hx:15: characters 3-12 : Abstract<Int> has no field named
test/Example.hx:16: characters 4-8 : Unknown identifier : name
test/Example.hx:16: characters 3-24 : String has no field toValue
test/Example.hx:17: characters 3-23 : String has no field toValue
test/Example.hx:18: characters 3-21 : String has no field named
```

**Comment by jasononeil** [comment]
Oh man, my fix to make it work on Haxe 3.3 broke 3.2 :( I might have to resort to conditional compilation to get it to work.

**Comment by francescoagati** [comment]
thanks works now :-)
